src/player.h

```diff
@@ -614,6 +615,8 @@ class player : public Character, public JsonSerializer, public JsonDeserializer
         void set_pain(int npain) override;
         /** Returns perceived pain (reduced with painkillers)*/
         int get_perceived_pain() const override;
+        /** Modifies perceived pain indirectly */
+        void mod_perceived_pain( std::function<void()> modifier );
```

Comment:
Passing a whole function and requiring `<functional>` in a header that is included everywhere just to represent 2 versions (set and mod) of pain function sound like an overkill to me.